broadway_api/utils/bootstrap.py

```diff
@@ -49,18 +90,110 @@ def initialize_course_tokens(settings, course_config_path):
         course_dao.insert_or_update(course)
 
 
-def initialize_database(settings):
-    uri = "mongodb://{}:{}".format(config.DB_HOST, config.DB_PORT)
+def initialize_database(settings: Dict[str, Any], flags: Dict[str, Any]):
+    logger.info("initializing database")
 
     try:
-        db_client = MongoClient(uri, serverSelectionTimeoutMS=5000)
+        db_client = MongoClient(
+            flags["mongodb_dsn"],
+            serverSelectionTimeoutMS=flags["mongodb_timeout"] * 1000,
+        )
         db_client.server_info()
+
+        settings["DB"] = db_client
+
+        dao = WorkerNodeDao(settings)
+        logger.info("resetting ws worker nodes")
+        dao.reset_worker_nodes()
+
     except ConnectionFailure as e:
         logger.critical("failed to connect to mongo server: {}".format(repr(e)))
         sys.exit(1)
 
-    settings["DB"] = MongoClient(uri)
 
-    dao = WorkerNodeDao(settings)
-    logger.info("resetting ws worker nodes")
-    dao.reset_worker_nodes()
+def initialize_signal_handler(settings: Dict[str, Any], flags: Dict[str, Any]):
+    logger.info("initializing signal handler")
+
+    def shutdown():
+        logger.info("shutting down")
+        tornado.ioloop.IOLoop.current().stop()
+
+    def handler(sig, frame):
+        tornado.ioloop.IOLoop.instance().add_callback_from_signal(shutdown)
```

Comment:
`tornado.ioloop.IOLoop.instance()` has been deprecated since version `5.0`. Please change this to `tornado.ioloop.IOLoop.current()`.